build/helper/documentation_helper.py

```diff
@@ -485,14 +485,16 @@ def format_type_for_rst_documentation(param, numpy, config):
     else:
         p_type = param['type_in_documentation']
 
-    if param['is_string'] is True:
-        p_type = 'str'
-    elif param['is_buffer'] is True and numpy is True:
-        p_type = 'numpy.array(dtype=numpy.{0})'.format(get_numpy_type_for_api_type(param['type'], config))
-    elif param['use_list'] is True:
-        p_type = 'list of ' + p_type
-    elif param['use_array'] is True:
-        p_type = 'array.array("{0}")'.format(get_array_type_for_api_type(param['type']))
+    # If type_in_documentation was set in metadata, we use it as is
+    if param['type_in_documentation_was_calculated'] or numpy:
+        if param['is_string'] is True:
+            p_type = 'str'
+        elif param['is_buffer'] is True and numpy is True:
```

Comment:
Is it possible for numpy to be True but `param['is_buffer']` is False? It seems to me like that would be a metadata mistake. I think we should capture that in code by removing `is_buffer` from the condition and making it an assert.